### DISCUSSION TITLE
Multiple DS18x20 with alphanumeric name - Berry autoexec.bat script

### DISCUSSION THREAD

**Discussion by jpier123** [arendst/Tasmota#19368] [General]
I'm looking for some help with getting ESP32's Berry autoexec script that could substitute System#Boot rule turning multile (x12) DS18b20 IDs into althanumeric name using DS18x20_USE_ID_ALIAS. I have simply no space left accross rule1..3.

####################################################################################
Tasmota 13.1.0.1 

 xsns_05_esp32_ds18x20.ino - DS18x20 temperature sensor support for Tasmota

#define DS18x20_USE_ID_ALIAS in my_user_config.h or user_config_override.h

* Use alias for fixed sensor name in scripts by autoexec. 
* Command: DS18Alias XXXXXXXXXXXXXXXX,N where XXXXXXXXXXXXXXXX full serial and N an alphanumeric value, the complete name is replaced with it, 
######################################################################################

Rule example: 

Rule1 ON System#Boot DO 
Backlog DS18Alias 783C01F096F2BB28,DS01;
DS18Alias 113C01F096174528,DS02;
DS18Alias 8B0316C310F3FF28,DS03;
...
DS18Alias 8B0316C310F3FF29,DS12 ENDON

Result in JSON:  "DS01":{"Id":"783C01F096F2BB28","Temperature":26.3} (example with N=DS01)...

Thanks in advance

JP

**Comment by barbudor** [comment]
You don't need Berry for that
As you have a filesystem, tasmota also supports 'autoexec.bat'. It's not a joke (or a bit because of the name of the file) but you can use the filesystem file editor to create a file named 'autoexec.bat'
The file should just have all you DS18Alias commands, one per line
Et voilà

With Berry, it would just be 'autoexec.be', having 1 per line
```
tasmota.cmd("DS18Alias xxxx x") 
```

**Reply by sfromis** [reply]
... the Backlog command to have multiple commands in one string, also works via `tasmota.cmd` in Berry, except for not getting result back into Berry.

One small difference from a `System#boot` trigger is that Berry `autoexec.be` executes earlier, already before network connection. If that is important, you can also use that trigger from Berry.
`tasmota.add_rule('System#boot', /->tasmota.cmd('Backlog ...')`

Using `autoexec.bat` is about as late as `System#boot`, so ....

**Reply by jpier123** [reply]
Hi Barbudor, Sfromis,

All sorted; Thank you both for your help. Two dedicated Berry scipts (autoexec.be and DS18Alias.be) did the job.

1.autoexec.be
load("DS18Alias.be")

2.DS18Alias.be
tasmota.cmd("DS18Alias B37FF0FF0264FF28,DS01")
tasmota.cmd("DS18Alias 1B37387B0E64FF28,DS02")


DEBUG LOG:

17:30:20.147 SRC: Berry
17:30:20.148 CMD: Grp 0, Cmd 'DS18ALIAS', Idx 1, Len 21, Pld -99, Data 'B37FF0FF0264FF28,DS01'
17:30:20.155 RSL: RESULT = {"DS01":{"Id":"B37FF0FF0264FF28"},"":{"Id":"1B37387B0E64FF28"}}
17:30:20.164 SRC: Berry
17:30:20.165 CMD: Grp 0, Cmd 'DS18ALIAS', Idx 1, Len 21, Pld 1, Data '1B37387B0E64FF28,DS02'
17:30:20.172 RSL: RESULT = {"DS01":{"Id":"B37FF0FF0264FF28"},"DS02":{"Id":"1B37387B0E64FF28"}}
17:30:20.180 BRY: Successfully loaded 'autoexec.be'
...
17:30:32.454 RSL: SENSOR = {"Time":"2023-08-23T17:30:32","Switch4":"ON","COUNTER":{"C1":0,"C2":0},"DS01":{"Id":"7FF0FF0264FF","Temperature":25.1},"DS02":{"Id":"37387B0E64FF","Temperature":25.3},

Best Regards,

JP
